### PR TITLE
Change to literalinclude and downloadable BibTeX citation  #18442

### PR DESCRIPTION
**Pull request by thorvaldj**
## PR Summary
An attempt in solving problem described in issue  #18442, by extracting BibTeX citation to a seperate file. 
The bib file is then included in the citation page when generating the docs. Also added a simple "download button" as described in the issue. 
Unsure about where to put the bib-file. Added it to the root of the docs along with the existing citing.rst, as this probably will be the only bib file for the documentation?

## PR Checklist

- [N/A] Has pytest style unit tests (and `pytest` passes).
- [N/A] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [N/A] New features are documented, with examples if plot related.
- [X] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [N/A] Conforms to Matplotlib style conventions (install `flake8-docstrings` and `pydocstyle<4` and run `flake8 --docstring-convention=all`).
- [N/A] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [N/A] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

**Review by story645 — APPROVED** [review]
Works beautifully! Thanks for the PR :smile:

File: doc/citing.rst [hunk 1]
```diff
@@ -8,22 +8,11 @@ please acknowledge this fact by citing `J. D. Hunter, "Matplotlib: A 2D
 Graphics Environment", Computing in Science & Engineering, vol. 9, no. 3,
 pp. 90-95, 2007 <https://doi.org/10.1109/MCSE.2007.55>`_.
 
-.. code-block:: bibtex
+.. literalinclude:: MCSE.2007.55.bib
```
Comment (jklymak) [inline doc/citing.rst:11]:
```suggestion
.. literalinclude:: MCSE.2007.55.bib
   :language: none
```
otherwise it gets python highlighting.

Comment (story645) [reply doc/citing.rst:11]: Then you should set the language to bib:
https://pygments.org/docs/lexers/#lexers-for-bibtex-bibliography-data-and-styles

Comment (thorvaldj) [reply doc/citing.rst:11]: Nice 👍  I set the language option as a new commit. Tested it, and the highlighting looks better. Both "bib", and "bibtex" keyword works. Went for the full name.

**Comment by QuLogic** [comment]
Thanks @thorvaldj! Congratulations on your first PR to Matplotlib :tada: We hope to hear from you again.

**Comment by thorvaldj** [comment]
> Thanks @thorvaldj! Congratulations on your first PR to Matplotlib 🎉 We hope to hear from you again.

Thank you! Glad to get the opportunity to contribute :-)

**Pull request merged**